test/client/bst-proxy-test.ts

```diff
@@ -5,6 +5,30 @@ import {Global} from "../../lib/core/global";
 import {BespokeServer} from "../../lib/server/bespoke-server";
 import {BSTProxy} from "../../lib/client/bst-proxy";
 
+describe("BSTProxy Programmatic", function () {
+    it("Starts and stops programmatically", function (done) {
+        let proxy = BSTProxy.http(9999).secretKey("SECRET_KEY");
+        proxy.start(() => {
+            assert.equal("SECRET_KEY", (proxy as any).bespokenClient.nodeID);
```

Comment:
surround with try {
} catch (e) {
  done(e);
}

to output error correctly instead of timing out.